tests/phpunit/Application/QueryStringParserTest.php

```diff
@@ -144,7 +146,7 @@ public function testHandlesSingleQuotedString(): void {
 	public function testHandlesOrQuotedStrings(): void {
 		$this->markTestSkipped( 'TODO' );
 
-		$parser = new QueryStringParser();
+		$parser = $this->newQueryStringParser();
```

Comment:
And this is why it's often worth creating that private method right away, even if initially there are no constructor arguments. Take note @JeroenDeDauw